sympy/core/numbers.py

```diff
@@ -222,6 +223,56 @@ def _literal_float(f):
     """Return True if n starts like a floating point number."""
     return bool(_floatpat.match(f))
 
+
+def _most_sig(x):
+    # return 0 if mpf precision < 1
+    if x[3] < 1:
+        return 0
+
+    x = (0,) + x[1:]
+
+    fx = mlib.to_float(x)
+    try:
+        decimal_digits = floor(log10(fx))
```

Comment:
I would compute the position of the most significant binary digit from the components of an mpf.
```
>>> x = Float('99901', '')._mpf_
>>> _, _, exp, bc = x    # extract exponent and bit count
>>> exp
0    # position of least significant bit of mantissa
>>> exp + bc - 1
16   # position of most significant binary digit
```